GDJS/Runtime/bitmapfontmanager.js

```diff
@@ -0,0 +1,265 @@
+  this.bitmapFont = this.patchInstalledBitmapFont(
+    PIXI.BitmapFont.from(bitmapFontStyle.fontFamily, bitmapFontStyle, {
+      chars: [
+        [' ', '~'], // All the printable ASCII characters
+      ],
+    })
+  );
+
+  this._defaultSlugFontName = this.bitmapFont.font;
+  return this.bitmapFont;
+};
+
+/**
+ * Update the resources data of the game. Useful for hot-reloading, should not be used otherwise.
+ * @param {ResourceData[]} resources The resources data of the game.
+ */
+gdjs.BitmapFontManager.prototype.setResources = function (resources) {
+  this._resources = resources;
+};
+
+/**
+ * When an object use an bitmapFont register the slug of the font and how many objects use it.
+ * @param {string} name Slug of the bitmapFont
+ */
+gdjs.BitmapFontManager.prototype.setFontUsed = function (name) {
+  this._fontUsed[name] = this._fontUsed[name] || { objectsUsingTheFont: 0 };
+  this._fontUsed[name].objectsUsingTheFont++;
+
+  const fontCache = this._fontTobeUnloaded[name];
+  if (fontCache) {
+    // The font is in the cache of unloaded font, because it was previously used and then marked as not used anymore.
+    // Remove it from the cache to avoid the font getting unloaded.
+    delete this._fontTobeUnloaded.name;
+  }
+};
+
+/**
+ * When an bitmapText object is removed, decrease the count of objects related to the font in the manager.
+ * When a font is not unused anymore, it goes in a temporary cache. The cache holds up to 10 fonts.
+ * If the cache reaches its maximum capacity, the oldest font is unloaded from memory.
+ * @param {string} name Slug of the bitmapFont
+ */
+gdjs.BitmapFontManager.prototype.removeFontUsed = function (name) {
+  if (!this._fontUsed[name]) {
+    // We tried to remove font that was never marked as used.
+    console.error(
+      'BitmapFont with name ' +
+        name +
+        ' was tried to be removed but was never marked as used.'
+    );
+    return;
+  }
+
+  this._fontUsed[name].objectsUsingTheFont--;
+
+  if (this._fontUsed[name].objectsUsingTheFont <= 0) {
+    delete this._fontUsed[name];
+
+    // Add the font name at the last position of the cache.
+    if (!this._fontTobeUnloaded.includes(name))
+      this._fontTobeUnloaded.push(name);
+
+    if (this._fontTobeUnloaded.length > 10) {
+      // Remove the first font (i.e: the oldest one)
+      const oldestUnloadedFont = this._fontTobeUnloaded.shift();
+      PIXI.BitmapFont.uninstall(oldestUnloadedFont);
+      console.log('Uninstall bitmapFont: ' + oldestUnloadedFont);
+    }
+  }
+};
+
+/**
+ * @param {string} bitmapFontResourceName
+ * @param {PIXI.Texture} texture
+ * @returns {PIXI.BitmapFont}
+ */
+gdjs.BitmapFontManager.prototype.getBitmapFontFromData = function (
+  bitmapFontResourceName,
+  texture
+) {
+  // Reuse an existing bitmapFont that is already in memory and already installed
+  for (let i = 0; i < this._loadedFontsData.length; ++i) {
```

Comment:
You iterating on an object, this is not guaranteed to work and will probably not because `.length` is undefined.
It's an object, so you need to pass a string as a `key` to `this._loadedFontsData[key]`. Here, it's probably the `bitmapFontResourceName`? 

We need to be very clear: this object is storing some "loaded font data". They are indexed by a key, which is ... (up to you to decide).